File: src/algorithms/remove-digit.js
/**
 * Given a string number representing a positive integer and a character digit,
 * returns the resulting string after removing exactly one occurrence of digit
 * from number such that the value of the resulting string in decimal form is
 * maximized.
 * 
 * @param {string} number - A string representing a positive number
 * @param {string} digit - A string representing a number digit
 * @returns {string} - Returns a maximized number in string format
 */
function removeDigit (number, digit) {
  let maxRes = '';
  for (let i = 0; i < number.length; i++) {
    if (number[i] === digit) {
      let temp = number.slice(0, i) + number.slice(i + 1);
      if (maxRes === '' || temp > maxRes) {
        maxRes = temp;
      }
    }
  }
  maxRes = String(maxRes);
  return maxRes;
}

File: src/algorithms/remove-digit.py
def removeDigit(number, digit):
    """You are given a string number representing a positive integer and a 
    character digit. Return the resulting string after removing exactly one 
    occurrence of digit from number such that the value of the resulting string 
    in decimal form is maximized. The test cases are generated such that digit 
    occurs at least once in number.

    Args:
        number (string): A string of digits.
        digit (string): A string digit.

    Returns:
        string: Returns a modified string.
    """
    res = "0"
    for i in range(len(number)):
        if number[i] == digit:
            temp = number[0:i] + number[i + 1:]
            res = str(max(int(res), int(temp)))
    return res
